frontend/src/citizen-frontend/children/sections/service-need-and-daily-service-time/AttendanceSummaryTable.tsx

```diff
@@ -0,0 +1,148 @@
+// SPDX-FileCopyrightText: 2017-2023 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import React, { useMemo, useState } from 'react'
+import styled from 'styled-components'
+
+import { useTranslation } from 'citizen-frontend/localization'
+import { Result, combine } from 'lib-common/api'
+import FiniteDateRange from 'lib-common/finite-date-range'
+import { AttendanceSummary } from 'lib-common/generated/api-types/children'
+import { ServiceNeedSummary } from 'lib-common/generated/api-types/serviceneed'
+import LocalDate from 'lib-common/local-date'
+import { UUID } from 'lib-common/types'
+import { useApiState } from 'lib-common/utils/useRestApi'
+import IconButton from 'lib-components/atoms/buttons/IconButton'
+import ErrorSegment from 'lib-components/atoms/state/ErrorSegment'
+import Spinner from 'lib-components/atoms/state/Spinner'
+import ListGrid from 'lib-components/layout/ListGrid'
+import { FixedSpaceRow } from 'lib-components/layout/flex-helpers'
+import { AlertBox } from 'lib-components/molecules/MessageBoxes'
+import { H3, Label } from 'lib-components/typography'
+import { Gap } from 'lib-components/white-space'
+import colors from 'lib-customizations/common'
+import { faChevronLeft, faChevronRight } from 'lib-icons'
+
+import { getAttendanceSummary } from './api'
+
+interface AttendanceSummaryTableProps {
+  childId: UUID
+  serviceNeedsResponse: Result<ServiceNeedSummary[]>
+}
+
+export default React.memo(function AttendanceSummaryTable({
+  childId,
+  serviceNeedsResponse
+}: AttendanceSummaryTableProps) {
+  const t = useTranslation()
+  const [attendanceSummaryDate, setAttendanceSummaryDate] = useState(() =>
+    LocalDate.todayInHelsinkiTz().startOfMonth()
+  )
+  const attendanceSummaryRange = useMemo(
+    () =>
+      new FiniteDateRange(
+        attendanceSummaryDate.startOfMonth(),
+        attendanceSummaryDate.lastDayOfMonth()
+      ),
+    [attendanceSummaryDate]
+  )
+  const [attendanceSummaryResponse] = useApiState(
+    () => getAttendanceSummary(childId, attendanceSummaryDate),
+    [childId, attendanceSummaryDate]
+  )
```

Comment:
the new way would be to use the query framework, but I don't know if this is deprecated either, fine by me personally